web/src/main/java/com/graphhopper/http/cli/ExportCommand.java

```diff
@@ -0,0 +1,234 @@
+package com.graphhopper.http.cli;
+
+import com.google.common.collect.Lists;
+import com.graphhopper.GraphHopper;
+import com.graphhopper.http.GraphHopperManaged;
+import com.graphhopper.http.GraphHopperServerConfiguration;
+import com.graphhopper.routing.ev.DecimalEncodedValue;
+import com.graphhopper.routing.ev.EnumEncodedValue;
+import com.graphhopper.routing.ev.RoadClass;
+import com.graphhopper.routing.util.AllEdgesIterator;
+import com.graphhopper.routing.util.CarFlagEncoder;
+import com.graphhopper.routing.util.DefaultFlagEncoderFactory;
+import com.graphhopper.routing.util.EncodingManager;
+import com.graphhopper.storage.GraphHopperStorage;
+import com.graphhopper.storage.NodeAccess;
+import com.graphhopper.swl.CustomGraphHopperOSM;
+import com.graphhopper.swl.StableIdEncodedValues;
+import com.graphhopper.util.DistanceCalcEarth;
+import com.graphhopper.util.FetchMode;
+import com.graphhopper.util.PointList;
+import io.dropwizard.cli.ConfiguredCommand;
+import io.dropwizard.setup.Bootstrap;
+import net.sourceforge.argparse4j.inf.Namespace;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Custom command used to export a GraphHopper street network in CSV format. Developed to mimic as possible the logic
+ * from R5's built-in street network export command.
+ *
+ * To run, the command expects an OSM file for the region to be specified with GraphHopper's
+ * -Ddw.graphhopper.datareader.file command line argument, and pre-built graph files (originating from the same OSM)
+ * to be present in the /transit_data/graphhopper subfolder.
+ *
+ * Example of calling this command:
+ * java -Xmx10g -Ddw.graphhopper.datareader.file=./region_cutout.osm.pbf \
+ * -jar web/target/graphhopper-web-1.0-SNAPSHOT.jar export config-usa.yml
+ */
+
+public class ExportCommand extends ConfiguredCommand<GraphHopperServerConfiguration> {
+    private static final Logger LOG = LoggerFactory.getLogger(ExportCommand.class);
+
+    private static final String FLAG_ENCODERS = "car,bike,foot";
```

Comment:
Note that these flag encoders are only used to initialize `osmTaggedGraphHopper` below, which is only responsible for parsing tags from OSM ways to calculate certain columns in the output CSV. 

This worked when `configuredGraphHopper` (below) was initialized using pre-built GH graph files generated with only car/bike/foot encoders, but if we build the initial GH graph with truck flag encoders also used, we should ensure that `osmTaggedGraphHopper` is still able to provide OSM tag info for every edge iterated over in [this loop below](https://github.com/replicahq/graphhopper/pull/19/files#diff-883eacff0fd02718e3c7744ca33a637eR116) even if it's only been created using the three flag encoders here.

Reply:
@michaz would love you input on this as well - when comparing the edges included in graphs built using just `car|bike|foot` vs graphs built using `car|bike|foot|truck`, would the latter include more edges than the former? Or does the addition of the truck flag encoder not result in edges being included that otherwise wouldn't be?

Reply:
I'm not sure, but I do think it's dangerous to rely on the two GraphHoppers here being in prestabilized harmony about their edge numbers.

Can we put another EncodedValue for the OSM ID into the schema of the _regular_ GraphHopper? Then we could just use your `CustomGraphHopperOSM` as an OSM DB, and it wouldn't be responsible for knowing about the _other_ GHs edge ids.

Alternatively, we could also put _all_ the information we need into the regular GraphHopper (some more EncodedValues).. then it would be an actual export and wouldn't need to re-read the OSM dump at all..